test/GPv2Settlement.test.ts

```diff
@@ -127,4 +129,87 @@ describe("GPv2Settlement", () => {
       ).to.be.revertedWith("invalid pair");
     });
   });
+
+  describe("uniswapTrade", () => {
+    async function tradeWithAddressOrder(
```

Comment:
This is unfortunate, we can't create a mock `IUniswapV2Pair` because we can't make it land on a specific address... Maybe `uniswapPairAddress` can be virtual and overloaded by `GPv2SettlementTestInterface` so that we can specify a mock pair address?